src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -48,8 +48,8 @@ private class ConfigurationCacheItem
             _featureFilters = featureFilters ?? throw new ArgumentNullException(nameof(featureFilters));
             _sessionManagers = sessionManagers ?? throw new ArgumentNullException(nameof(sessionManagers));
             _logger = loggerFactory.CreateLogger<FeatureManager>();
-            _filterMetadataCache = new ConcurrentDictionary<string, IFeatureFilterMetadata>();
-            _contextualFeatureFilterCache = new ConcurrentDictionary<string, ContextualFeatureFilterEvaluator>();
+            _filterMetadataCache = new ConcurrentDictionary<ValueTuple<string, Type>, IFeatureFilterMetadata>();
+            _contextualFeatureFilterCache = new ConcurrentDictionary<ValueTuple<string, Type>, ContextualFeatureFilterEvaluator>();
```

Comment:
Instead of caching type, I recommend using just the name of the type like  we did before. Perhaps `Tuple<string, string>``

Reply:
I convinced myself that using $"{filterName}{Environment.NewLine}{appContextType.FullName}" as the cache key is better.
I plan to follow the previous way.

Reply:
But I don't see the change?